src/stdlib_hashmap_chaining.f90

```diff
@@ -0,0 +1,848 @@
+        type(key_type), intent(in)                  :: key
+
+        integer(int_hash)                      :: hash_val, hash_index
+        type(chaining_map_entry_type), pointer :: gentry, pentry, sentry
+
+        if ( map % probe_count > inmap_probe_factor * map % call_count ) then
+            if ( map % nbits < max_bits .AND. &
+                 map % num_entries > size( map % slots, kind=int_index ) ) then
+                call expand_slots(map)
+            end if
+        end if
+        map % call_count = map % call_count + 1
+        hash_val = map % hasher( key )
+        hash_index = fibonacci_hash( hash_val, map % nbits )
+        pentry => map % slots(hash_index) % target
+        sentry => pentry
+
+        climb_chain: do
+            gentry => pentry
+            map % probe_count = map % probe_count + 1
+            if (.not. associated( gentry ) ) then
+                write(error_unit,*) "gentry not associated"
+                inmap = 0
+                return
+            else if ( hash_val == gentry % hash_val ) then
+                if ( key == gentry % key ) then
+! The swap to front seems to confuse gfortran's pointers
+!                    if ( .not. associated( pentry, sentry ) ) then
+!                    ! swap to front
+!                        pentry => gentry % next
+!                        gentry % next => sentry
+!                        sentry => gentry
+!                    end if
+                    inmap = gentry % inmap
+                    return
+                end if
+            end if
+            pentry => gentry % next
+        end do climb_chain
+
+    end subroutine in_chain_map
+
+
+    module subroutine init_chaining_map( map,        &
+                                         hasher,     &
+                                         slots_bits, &
+                                         status )
+!! Version: Experimental
+!!
+!! Routine to allocate an empty map with HASHER as the hash function,
+!! 2**SLOTS_BITS initial SIZE(map % slots), and SIZE(map % slots) limited
+!! to a maximum of 2**MAX_BITS. All fields are initialized.
+!! Arguments:
+!!     map         - the chaining hash map to be initialized
+!!     hasher      - the hash function to be used to map keys to slots
+!!     slots_bits - the bits of two used to initialize the number of slots
+!!     status      - an integer error status flag with the allowed values:
+!!         success - no problems were found
+!!         alloc_fault - map % slots or map % inverse could not be allocated
+!!         array_size_error - slots_bits is less than default_bits or
+!!             greater than max_bits
+!
+        class(chaining_hashmap_type), intent(out)  :: map
+        procedure(hasher_fun)                      :: hasher
+        integer, intent(in), optional              :: slots_bits
+        integer(int32), intent(out), optional      :: status
+
+        character(256)          :: errmsg = ''
```

Comment:
It will, but since the program is terminated when this is used, it won't matter much.
Anyway, I don't see why this can't be changed (only if the `allocate` leave the `errmsg` unchanged)

```suggestion
        character(256)          :: errmsg
```